assets/src/scripts/components/median-water-level-table/index.js

```diff
@@ -0,0 +1,117 @@
+        return `${total}<td class="${name}"></td>`;
+    }, '');
+    const options = {
+        valueNames: valueNames,
+        item: `<tr>${item}</tr>`,
+        page: 30,
+        pagination: true
+    };
+    new List('median-water-levels-div', options, samples);
+
+    return tbody;
+};
+
+/*
+ * Renders the median water level table
+ * @param  {Object} store               Redux store
+ * @param  {Object} node                DOM node to draw graph into
+ * @param  {Object} options.agencyCode  Agency of site to draw
+ * @param  {Object} options.siteId      ID of site to draw
+ * @param  {String} options.id          Unique ID for this component
+ */
+const drawTable = (store, node, agencyCode, siteId) => {
+    const component = select(node);
+    const table = component
+        .select('#median-water-levels-div')
+        .append('table')
+        .attr('id', 'median-water-levels-table')
+        .classed('usa-table', true);
+    component.select('#median-water-levels-div')
+        .append('ul')
+        .classed('pagination', true);
+
+    table.append('thead')
+         .append('tr')
+         .selectAll('th')
+         .data(COLUMN_HEADINGS)
+         .enter()
+         .append('th')
+         .text((col) => col);
+
+    table.call(link(store, (elem, {isRendered, waterLevels}) => {
+        // Add code to rendered
+        if (isRendered) {
+            drawTableBody(elem, waterLevels);
+        }
+    }, createStructuredSelector({
+        isRendered: isTableRendered,
+        waterLevels: getSiteMedianWaterLevels(agencyCode, siteId)
+    })));
+};
+
+
+/*
+ * Checks that data has been fetched.
+ * @param  {Object} store               Redux store
+ * @param  {Object} node                DOM node to draw graph into
+ * @param  {Object} options.agencyCode  Agency of site to draw
+ * @param  {Object} options.siteId      ID of site to draw
+ * @param  {String} options.id          Unique ID for this component
+ */
+const ensureDatafetch = (store, node, agencyCode, siteId) => {
+    // If a request for this site hasn't been made yet,
+    // make the water level service call and render the table elements.
+    if (!getMedianWaterLevelStatus(agencyCode, siteId)(store.getState())) {
+        store.dispatch(retrieveMedianWaterLevels(agencyCode, siteId));
+    }
+};
+
+/*
+ * Renders the median water level table - on button click
+ * @param  {Object} store               Redux store
+ * @param  {Object} node                DOM node to draw graph into
+ * @param  {Object} options.agencyCode  Agency of site to draw
+ * @param  {Object} options.siteId      ID of site to draw
+ * @param  {String} options.id          Unique ID for this component
+ */
+export default function(store, node, {agencyCode, siteId}) {
+    const component = select(node);
+    drawTable(store, node, agencyCode, siteId);
+    component.select('button').on('click', () => {
+        ensureDatafetch(store, node, agencyCode, siteId);
+        store.dispatch(renderTable());
```

Comment:
```suggestion
        store.dispatch(setTableRendered());
```

I'd suggest renaming this function to make it clear it's setting state rather than rendering to the DOM.

Reply:
Or maybe instead, since it will still be true after closing:

```suggestion
        store.dispatch(setTableWasExpanded());
```

Reply:
interesting suggestion. I will attempt it.